src/status_im/ui/screens/chat/components/input.cljs

```diff
@@ -73,27 +77,102 @@
+       :text-align-vertical    :center
+       :multiline              true
+       :default-value          text-value
+       :editable               (not cooldown-enabled?)
+       :blur-on-submit         false
+       :auto-focus             false
+       :on-focus               #(set-active-panel nil)
+       :max-length             chat.constants/max-text-size
+       :placeholder-text-color (:text-02 @colors/theme)
+       :placeholder            (if cooldown-enabled?
+                                 (i18n/label :cooldown/text-input-disabled)
+                                 (i18n/label :t/type-a-message))
+       :underlineColorAndroid  :transparent
+       :auto-capitalize        :sentences
+       :selection
+       (when (and cursor platform/android?)
+         (clj->js
+          {:start cursor
+           :end   cursor}))
+
+       :on-selection-change
+       (fn [_]
+         (when cursor
+           (re-frame/dispatch [::mentions/clear-cursor])))
+
+       :on-change
+       (fn [args]
+         (let [text (.-text ^js (.-nativeEvent ^js args))]
+           (on-text-change text)
+           (when platform/ios?
+             (re-frame/dispatch [::mentions/calculate-suggestions mentionable-users]))))
+
+       :on-text-input
+       (fn [args]
+         (let [native-event  (.-nativeEvent ^js args)
+               text          (.-text ^js native-event)
+               previous-text (.-previousText ^js native-event)
+               range         (.-range ^js native-event)
+               start         (.-start ^js range)
+               end           (.-end ^js range)]
+           (re-frame/dispatch
+            [::mentions/on-text-input
+             {:new-text      text
+              :previous-text previous-text
+              :start         start
+              :end           end}])
+           (when platform/android?
+             (re-frame/dispatch [::mentions/calculate-suggestions mentionable-users]))))}]]))
+
+(defn mention-item
+  [[_ {:keys [identicon alias name] :as user}]]
+  (let [title name
+        subtitle? (not= alias name)]
+    [list-item/list-item
+     (cond-> {:icon [rn/view {:style {}}
+                     [rn/image
+                      {:source      {:uri identicon}
+                       :style       (photo-style/photo-border
+                                     photo-style/default-size
+                                     nil)
+                       :resize-mode :cover}]]
+              :icon-container-style {}
+              :size :small
+              :text-size :small
+              :title title
+              :on-press
+              (fn []
+                (re-frame/dispatch [:chat.ui/select-mention user]))}
+
+       subtitle?
+       (assoc :subtitle alias))]))
+
+(defn autocomplete-mentions []
+  (let [suggestions @(re-frame/subscribe [:chat/mention-suggestions])]
+    (when (seq suggestions)
+      (let [height (* 52 (min 4.5 (count suggestions)))]
```

Comment:
why 4.5 ? will it work on small screens ? probably we could count this number depending on the screen height

Reply:
that's a good point, can it be percentage based? I know I said 4.5 but this might indeed overflow on smaller screens

Reply:
> probably we could count this number depending on the screen height

we probably can also take in count screen height and keyboard height, but I would address this when it is proven to be an issue. Meanwhile I will just change it to 3.5

for 4.5 see https://github.com/status-im/status-react/pull/11122#issuecomment-685700952

Reply:
> that's a good point, can it be percentage based?

it can be either fixed number or based on keyboard/screen height, but I'm still not convinced that this is a big problem and would revisit it after this PR. It is big and a bit complicated already.

Reply:
4.5 works fine on small device

Reply:
then we are good for now I think, thanks @flexsurfer